static/js/settings.js

```diff
@@ -38,20 +38,23 @@
             server_storage: labelCollectionCheckbox.checked,
         };
 
-        const response = await apiRequest(
-            `/profiles/${profileId}/`,
-            {
-                method: "PATCH",
-                body: JSON.stringify(settings),
-            },
-        );
-        if (response.ok) {
+        try {
+            const response = await apiRequest(
+                `/profiles/${profileId}/`,
+                {
+                    method: "PATCH",
+                    body: JSON.stringify(settings),
+                },
+            );
+            if (!response.ok) {
+                throw new Error("Immediately catch'd to show an error message.");
```

Comment:
That's exactly correct! Requests can either be rejected by the server (resulting in `response.ok` being `false`), or encountering a network error (throws an error (aka rejects the `fetch` Promise)). Since I only added a single error message for both cases, I used this to make sure a server rejection ends up in the same code path as a network rejection when it comes to showing an error message to the user (`saveError.classList.remove("hidden");`).

We could of course also create separate error messages (e.g. "Please try again" vs. "Check your network connection"), but given how in most cases the resolution is to either try again or give up, how I was already coming up with content strings myself, and to minimise the localisation burden, I went with the easy way and just a single error message :)